src/services/distributions.ts

```diff
@@ -1,71 +1,70 @@
-import {CatalogService} from './catalog';
 import * as Joi from '@hapi/joi';
 import Pino from 'pino';
 import {QueryResult, QueryTermsService} from './query';
+import {Catalog} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
 
 export interface ConstructorOptions {
   logger: Pino.Logger;
+  catalog: Catalog;
 }
 
 const schemaConstructor = Joi.object({
   logger: Joi.object().required(),
+  catalog: Joi.object().required(),
 });
 
 export interface QueryOptions {
-  distributionId: string;
+  source: string;
   query: string;
 }
 
 const schemaQuery = Joi.object({
-  distributionId: Joi.string().required(),
+  source: Joi.string().required(),
   query: Joi.string().required(),
 });
 
 export interface QueryAllOptions {
-  distributionIds: string[];
+  sources: string[];
   query: string;
 }
 
 const schemaQueryAll = Joi.object({
-  distributionIds: Joi.array().items(Joi.string().required()).min(1).required(),
+  sources: Joi.array().items(Joi.string().required()).min(1).required(),
   query: Joi.string().required(),
 });
 
 export class DistributionsService {
   protected logger: Pino.Logger;
-  protected catalogService: CatalogService;
+  protected catalog: Catalog;
 
   constructor(options: ConstructorOptions) {
     const args = Joi.attempt(options, schemaConstructor);
     this.logger = args.logger;
-    this.catalogService = new CatalogService({logger: args.logger});
+    this.catalog = args.catalog;
   }
 
   async query(options: QueryOptions): Promise<QueryResult> {
     const args = Joi.attempt(options, schemaQuery);
-    this.logger.info(`Preparing to query source "${args.distributionId}"...`);
-    const accessService = await this.catalogService.getAccessServiceByDistributionId(
-      args.distributionId
-    );
-    if (accessService === null) {
+    this.logger.info(`Preparing to query source "${args.source}"...`);
+    const dataset = await this.catalog.getByIdentifier(args.source);
+    if (dataset === undefined) {
       throw Error(
-        `Access service of source "${args.distributionId}" not found in catalog`
+        `Dataset with identifier "${args.identifier}" not found in catalog`
```

Comment:
This throws the error underneath if a non-existing source is given:

`Dataset with identifier "undefined" not found in catalog`

Also, perhaps change the word `Dataset` to `Source`?